# Curio (python) mobile base controller node parameters

# Servo driver port settings
# port: /dev/cu.wchusbserialfd5110
port: /dev/ttyUSB0
baudrate: 115200
timeout: 1.0

# Update frequencies: control loop
control_frequency: 20.0

# Robot dimensions and limits
max_linear_velocity:  0.37   # [m/s]
max_angular_velocity: 1.45   # [rad/s]

#  ROS frame convention for mobile robots:
# 
#                         x
#                         ^
#                         | longatudinal dir
#                         |
#  lateral dir  y <-------
# 
# 
# Wheel geometry / servo location. The numbers are servo serial ids:
# two digits for wheel servos, three digits for steering servos.
# 
#     <- front wheel lat separation ->
#    11 / 111 |-----|-----| 21 / 212   ---
#                   |                   ^
#                   |             front wheel lon separation
#       <- mid wheel lat separation ->  v
#        12 |-------|-------| 22       ---
#                   |                   ^
#                   |             back wheel lon separation
#                   |                   v
#   13 / 131 |------|------| 23 / 233  ---
#      <- back wheel lat separation ->
# 
wheel_radius: 0.060
mid_wheel_lat_separation: 0.52
front_wheel_lat_separation: 0.47 
front_wheel_lon_separation: 0.28
back_wheel_lat_separation: 0.47
back_wheel_lon_separation: 0.25

# Wheel servo serial id mappings
wheel_servos: [
    {
        id: 11,
        lon_label: front,
        lat_label: left,
    },
    {
        id: 12,
        lon_label: mid,
        lat_label: left,
    },
    {
        id: 13,
        lon_label: back,
        lat_label: left,
    },
    {
        id: 21,
        lon_label: front,
        lat_label: right,
    },
    {
        id: 22,
        lon_label: mid,
        lat_label: right,
    },
    {
        id: 23,
        lon_label: back,
        lat_label: right,
    },
]

# Steer servo serial id mappings and angle adjustments. The offsets
# should be set so that the corner wheel rotation axis is parallel to
# the rover's y-axis when the commanded angular velocity is zero. 
steer_servos: [
    {
        id: 111,
        lon_label: front,
        lat_label: left,
        offset: -10,
    },
    {
        id: 131,
        lon_label: back,
        lat_label: left,
        offset: -5,
    },
    {
        id: 211,
        lon_label: front,
        lat_label: right,
        offset: 40,
    },
    {
        id: 231,
        lon_label: back,
        lat_label: right,
        offset: 20,
    },
]

